packages/rainbowkit/src/components/RainbowKitProvider/AuthenticationContext.tsx

```diff
@@ -55,7 +60,7 @@ export function RainbowKitAuthenticationProvider<Message = unknown>({
 }: RainbowKitAuthenticationProviderProps<Message>) {
   // When the wallet is disconnected, we want to tell the auth
   // adapter that the user session is no longer active.
-  useAccount({
+  const { connector } = useAccount({
     onDisconnect: () => {
       adapter.signOut();
     },
```

Comment:
@KosmosKey Wagmi also has an optional `onConnect` callback here. That might simplify this if it is fired when the address changes